pkg/cmd/server/api/types.go

```diff
@@ -150,9 +149,120 @@ type OAuthConfig struct {
 	// AssetPublicURL is used for building valid client redirect URLs for external access
 	AssetPublicURL string
 
-	// all the handlers here
+	//IdentityProviders is an ordered list of ways for a user to identify themselves
+	IdentityProviders []IdentityProvider
+
+	// GrantConfig describes how to handle grants
+	GrantConfig GrantConfig
+
+	// SessionConfig hold information about configuring sessions.
+	SessionConfig *SessionConfig
+
+	TokenConfig TokenConfig
 }
 
+type TokenConfig struct {
+	// Max age of authorize tokens
+	AuthorizeTokenMaxAgeSeconds int32
+	// Max age of access tokens
+	AccessTokenMaxAgeSeconds int32
+}
+
+type SessionConfig struct {
+	// SessionSecrets list the secret(s) to use to encrypt created sessions. Used by AuthRequestHandlerSession
+	SessionSecrets []string
```

Comment:
open a followup issue for me to make this structured. secrets have to be in pairs (one for signing, one for encryption, with specific length requirements on the encryption key)